src/custom/hooks/useAllTransactionsAndOrders.ts

```diff
@@ -0,0 +1,59 @@
+import { useMemo } from 'react'
+import { useAllTransactions } from 'state/transactions/hooks'
+import { useOrders } from 'state/orders/hooks'
+import { useActiveWeb3React } from 'hooks'
+import { Order, OrderStatus } from 'state/orders/actions'
+import { TransactionDetails } from '@src/state/transactions/reducer'
+
+type TransactionAndOrder =
+  | (Order & { addedTime: number })
+  | (TransactionDetails & {
+      id: string
+      status: 'pending' | 'fulfilled'
+    })
+
+/**
+ * Returns whether a transaction/order happened in the last day (86400 seconds * 1000 milliseconds / second)
+ * @param time to check for recency
+ */
+export function isRecent(time: number): boolean {
+  return new Date().getTime() - time < 86_400_000
+}
+
+function isPending(data: TransactionAndOrder) {
+  return data.status === OrderStatus.PENDING
+}
+
+function isFulfilled(data: TransactionAndOrder) {
+  return data.status === OrderStatus.FULFILLED
+}
+
+export default function useAllTransactionsAndOrders() {
+  const { chainId } = useActiveWeb3React()
+  const allTransactions = useAllTransactions()
+  const allNonEmptyOrders = useOrders({ chainId })
+
+  return useMemo(() => {
+    const adjustedTransactions = Object.values(allTransactions)
+      .map(tx => ({ ...tx, id: tx.hash, status: tx.receipt ? 'fulfilled' : 'pending' }))
+      .filter(tx => isRecent(tx.addedTime))
```

Comment:
In general i feel this hook has a lot of logic, right. Is his responsibility for example remove old orders, or there should be a maintenance tasks updating the state?